node/services/service.go

```diff
@@ -36,13 +36,13 @@ func HeaderSyncer(
 }
 
 // P2PSubscriber creates a new header.P2PSubscriber.
-func P2PSubscriber(lc fx.Lifecycle, sub *pubsub.PubSub) *header.P2PSubscriber {
+func P2PSubscriber(lc fx.Lifecycle, sub *pubsub.PubSub) (*header.P2PSubscriber, *header.P2PSubscriber) {
```

Comment:
Why returning it twice?

Reply:
Good Q. According to docs: https://github.com/uber-go/fx/blob/master/annotated.go#L228-L234.

So if you return the same thing as two different types, you have to return it twice. We didn't have to do so with `fxutil` before, as I made this a special case there. There is no special case in FX, and I also feel why they do so, so we have what we have.

Reply:
ack